### PR TITLE
Merge to commit before monorepo change, reimplemented

### PR DESCRIPTION
**Pull request by amgleitman**
#### Please select one of the following
- [x] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

This is a reimplementation of #1795. It merges up to 1c1dfab, one commit before [the mono repo commit](714b502b0c7a5f897432dbad388c02d3b75b4689).

## Changelog

[GENERAL] [ADDED] - Merge to commit before monorepo change

## Test Plan

The notable items in #1795 have been gone through carefully, and things still work.

* `setColorScheme` works fine on macOS Paper, but it appears to be wonky on macOS Fabric. However, considering that macOS Fabric is still highly experimental, I think this is okay for now.
* The new `RNTLegacyView` test page works on both macOS and iOS Fabric.

### CODE REVIEW

**Comment by Saadnajmi** [comment]
From the Old PR I had, I see I wrote 
"Flatlist moved to it's own package"

I think we'll need to publish a "@react-native-macos/flatlist" package because we have diffs in there, so we don't want to accidentally use the core one. I think I didn't do anything for that in my old PR. Did you do anything to that regard?

**Comment by amgleitman** [comment]
> From the Old PR I had, I see I wrote "Flatlist moved to it's own package"
> 
> I think we'll need to publish a "@react-native-macos/flatlist" package because we have diffs in there, so we don't want to accidentally use the core one. I think I didn't do anything for that in my old PR. Did you do anything to that regard?

No I didn't. I'll take a closer look at these diffs.

**Comment by amgleitman** [comment]
> Looks like the RNTA pipeline isn't testing against newly published packages. -- It should do something similar to the init pipeline, where it starts a verdaccio server, publishes the new bits to that, then runs RNTA against that server to verify the bits in the PR, rather than the bits on NPM.

After experimenting with the Integration pipeline for a few days, it's clear that this fix will be rather involved with react-native-macos becoming more like a monorepo. Since @tido64 should know more about react-native-test-app than I do, I think he would be a valuable resource.

For now, we can probably press forward with this anyway. We already test our ability to use this in apps with react-native-macos-init and RNTester, so that validation may be enough.

**Comment by tido64** [comment]
I'd be happy to help when you think the repo is in a better state. Just an FYI, you probably want to bump the minor version and not point at a local `.tgz` file here:

https://github.com/microsoft/react-native-macos/blob/ebf12739065db35cb4d818a35370c8f119fc60d5/.ado/apple-integration.yml#L53-L54

https://github.com/microsoft/react-native-macos/blob/ebf12739065db35cb4d818a35370c8f119fc60d5/.ado/apple-integration.yml#L70-L71

File: packages/virtualized-lists/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "@react-native-mac/virtualized-lists",
```
Comment (Saadnajmi) [inline packages/virtualized-lists/package.json:2]:
```suggestion
  "name": "@react-native-macos/virtualized-lists",
```

Why "mac"?

Comment (amgleitman) [reply packages/virtualized-lists/package.json:2]: Somebody else is squatting on the @react-native-macos scope on NPMJS. We're filing a dispute for it, but we don't have an estimate for when it'll be resolved.

File: Libraries/Components/TextInput/TextInput.d.ts [hunk 2]
```diff
@@ -321,31 +331,31 @@ export interface TextInputMacOSProps {
   /**
    * Fired when a supported element is pasted
    */
-  onPaste?: (event: PasteEvent) => void | undefined;
+  onPaste?: ((event: PasteEvent) => void) | undefined; // [macOS]
```
[the remark this answers is not in the document]
Comment (amgleitman) [reply Libraries/Components/TextInput/TextInput.d.ts:334]: Okay, wasn't sure.

File: packages/virtualized-lists/package.json (hunk 1 above)
Comment (Saadnajmi) [reply packages/virtualized-lists/package.json:2]: What? Who?

Comment (Saadnajmi) [reply packages/virtualized-lists/package.json:2]: Talked offline. Let's please make sure we track this.. I would want that namespace long term.

**Review by Saadnajmi — APPROVED** [review]

File: packages/virtualized-lists/package.json (hunk 1 above)
Comment (amgleitman) [reply packages/virtualized-lists/package.json:2]: Will do.

**Pull request merged**